dashboard/app/models/script.rb

```diff
@@ -206,8 +206,21 @@ def old_professional_learning_course?
   def generate_plc_objects
     if old_professional_learning_course?
       unit_group = UnitGroup.find_by_name(professional_learning_course)
-      unless unit_group
-        unit_group = UnitGroup.new(name: professional_learning_course)
+      if unit_group
+        # Check if anything needs to be updated on the PL course
+        unit_group.published_state = get_published_state if unit_group.published_state != get_published_state
+        unit_group.instruction_type = get_instruction_type if unit_group.instruction_type != get_instruction_type
+        unit_group.participant_audience = get_participant_audience if unit_group.participant_audience != get_participant_audience
+        unit_group.instructor_audience = get_instructor_audience if unit_group.instructor_audience != get_instructor_audience
```

Comment:
are the `if unit_group.foo != foo` clauses actually doing anything, given you already have the `if unit_group.changed?` check below? my suggestion would be to add test cases covering all the behavior you care about, and then consider removing the `if` clauses on lines 211-214 if you can do so without breaking the tests.